tests/pos/t284-pos.scala

```diff
@@ -1,5 +1,5 @@
 trait B[T] {
-    def f1(a: T): Unit { }
+    //def f1(a: T): Unit { }
```

Comment:
Yes, this should be a separate issue, not a silent commenting out of tests. 
And why will ExtensionMethods assume that refinement of unit is a value class. I would that this is a bug.